pkg/broker/broker_test.go

```diff
@@ -0,0 +1,88 @@
+package broker
```

Comment:
shouldn't this be in /test, not /pkg ?

Reply:
Both ways are acceptable in my book.
The reason I did it like this is cause `decodePlan()` isn't exported. So we will have to export it if we want to test it from a separate package.

I prefer when unit tests are part of the package.